force-app/main/default/lwc/participantAdder/participantAdder.js

```diff
@@ -0,0 +1,142 @@
+import BACK_WARNING_LABEL from "@salesforce/label/c.Clicking_Back_Button_In_Modal_Warning";
+import CANCEL_LABEL from "@salesforce/label/c.Cancel";
+import LOADING_LABEL from "@salesforce/label/c.Loading";
+import SERVICE_FIELD from "@salesforce/schema/ServiceSchedule__c.Service__c";
+import SAVE_SUCCESS from "@salesforce/label/c.Add_Service_Participants_Success";
+
+export default class ParticipantAdder extends LightningElement {
+    @api recordId;
+    errorMessage;
+    isLoaded = false;
+    serviceScheduleModel;
+    selectedParticipantContactIds;
+    existingParticipants;
+    labels = {
+        save: SAVE_LABEL,
+        success: SUCCESS_LABEL,
+        backWarning: BACK_WARNING_LABEL,
+        cancel: CANCEL_LABEL,
+        loading: LOADING_LABEL,
+        successMessage: SAVE_SUCCESS,
+    };
+
+    @wire(getRecord, {
+        recordId: "$recordId",
+        fields: [SERVICE_FIELD],
+    })
+    wiredSchedule(result) {
+        if (!(result.data || result.error)) {
+            return;
+        }
+        if (result.data) {
+            this.serviceId = result.data.fields[SERVICE_FIELD.fieldApiName].value;
+        }
+    }
+
+    @wire(getServiceScheduleModel, { recordTypeId: null })
+    wireServiceScheduleModel(result) {
+        if (!result) {
+            return;
+        }
+
+        if (result.data) {
+            this.serviceScheduleModel = JSON.parse(JSON.stringify(result.data));
+            this.extractLabels(this.serviceScheduleModel.labels.serviceParticipant);
+            this.isLoaded = true;
+        } else if (result.error) {
+            console.log(JSON.stringify(result.error));
+        }
+    }
+
+    extractLabels(labels) {
+        if (!labels) {
+            return;
+        }
+
+        Object.keys(labels).forEach(label => {
+            let value = labels[label];
+            this.labels[label] = value;
+        });
+    }
+
+    @wire(getExistingParticipantContactIds, { scheduleId: "$recordId" })
+    wiredExistingParticipants(result) {
+        if (!result) {
+            return;
+        }
+        this.existingParticipants = result; //cache for refreshing
+
+        if (result.data) {
+            this.selectedParticipantContactIds = result.data;
+        }
+    }
+
+    handleSave() {
+        let participantSelector = this.template.querySelector("c-participant-selector");
+
+        if (
+            !(
+                participantSelector &&
+                participantSelector.newParticipantsProgramEngagements
+            )
+        ) {
+            return;
+        }
+
+        saveNewParticipants({
+            engagements: participantSelector.newParticipantsProgramEngagements,
+            scheduleId: this.recordId,
+        })
+            .then(() => {
+                this.showSuccessToast(
+                    participantSelector.newParticipantsProgramEngagements.length
+                );
+                refreshApex(this.existingParticipants);
```

Comment:
Is this necessary if we are doing a refresh on the page in aura?

Reply:
Yes, sadly. It refreshes the view but keeps context of the wire.